File: 剑指offer/53.最大子数组和.cpp
class Solution {
 public:
    int maxSubArray(vector<int>& nums) {
      size_t len=nums.size();
      int ans=0,pre=nums[0];
      for(size_t i=1;i<len;i++){
        int cur = nums[i];
        pre=max(cur,pre+cur);
        ans=max(ans,pre);
      }
      return ans;
    }
};

File: 序号/53. 最大子数组和.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n=len(nums)
        dp=[0]*(n+1)
        ans=-math.inf
        for i ,x in enumerate(nums):
            dp[i+1]=max(x,x+dp[i])
            ans=max(ans,dp[i+1])
        return ans